torch/distributions/gamma.py

```diff
@@ -54,3 +55,12 @@ def log_prob(self, value):
     def entropy(self):
         return (self.alpha - torch.log(self.beta) + torch.lgamma(self.alpha) +
                 (1.0 - self.alpha) * torch.digamma(self.alpha))
+
+
+@register_kl(Gamma, Gamma)
+def kl_gamma_gamma(p, q):
```

Comment:
Why should this be public?

Reply:
The main reason is to allow users to use these implementations for user defined distributions. E.g.
```py
class LogGamma(Distribution):
    def __init__(self, alpha, beta):
        self._gamma = Gamma(alpha, beta)
    def rsample(self):
        return self._gamma.rsample().log()

@register_kl
def kl_loggamma_loggamma(p, q):
    return kl_gamma_gamma(p._gamma, q._gamma)
```
If the `_kl_gamma_gamma()` were private, then `kl_loggamma_loggamma` would need to use `kl_divergence` under the hood, and we'd be hitting the registry twice each call.

Reply:
Hmm I still think it would just be nicer to have one consistent API instead of a bunch of methods scattered over files. How about this: you can either use `distributions.kl`, or we add an option to lookup functions for types, and you can cache the function wherever you want. Something like:

```python
kl_gamma_gamma = distributions.kl(Gamma, Gamma)
@register_kl(LogGamma, LogGamma)
def kl_loggamma_loggamma(p, q):
    return kl_gamma_gamma(p._gamma, q._gamma)
```

Reply:
I think that is what @fritzo discussed in the distributions channel recently. I also think it would be better to have all of them in one place.

Reply:
@apaszke Good point, I've added a `kl_function` lookup method as you suggest.